sunpy/net/dataretriever/sources/goes.py

```diff
@@ -269,9 +269,9 @@ def search(self, *args, **kwargs):
         matchdict = self._get_match_dict(*args, **kwargs)
         req_wave = matchdict.get('Wavelength', None)
         if req_wave is not None:
-            wmin = req_wave.min.to(u.Angstrom, equivalencies=u.spectral())
-            wmax = req_wave.max.to(u.Angstrom, equivalencies=u.spectral())
-            req_wave = a.Wavelength(wmin, wmax)
+            wmin_round = round(req_wave.min.value) * u.Angstrom
+            wmax_round = round(req_wave.max.value) * u.Angstrom
```

Comment:
I still don't think this is correct? What if the user specifies a wavelength in GHz?

Reply:
To elaborate, `a.Wavelength()` is angstroms only if the argument has units equivalent to length.  It can alternatively be in kHz or keV.  So, `a.Wavelength(171*u.keV)` would also be misinterpreted by the change in this PR.

Reply:
I too believe this might not be the correct method 

how about round it after using u.spectral?

```
            wmin = round(req_wave.min.to(u.Angstrom, equivalencies=u.spectral()).value) * u.Angstrom
            wmax = round(req_wave.max.to(u.Angstrom, equivalencies=u.spectral()).value) * u.Angstrom
            req_wave = a.Wavelength(wmin, wmax)
```

Reply:
AStropy units has a round